### PR TITLE
Add extra codecs to allow custom keys

### PR DESCRIPTION
**Pull request by joeabbey**

### CODE REVIEW

File: client/encoding.go [hunk 1]
```diff
@@ -19,15 +23,27 @@ type Codec struct {
...
 	modBasic := module.NewBasicManager(moduleBasics...)
 	encodingConfig := MakeCodecConfig()
 	std.RegisterLegacyAminoCodec(encodingConfig.Amino)
 	std.RegisterInterfaces(encodingConfig.InterfaceRegistry)
 	modBasic.RegisterLegacyAminoCodec(encodingConfig.Amino)
 	modBasic.RegisterInterfaces(encodingConfig.InterfaceRegistry)
-	ethcodec.RegisterInterfaces(encodingConfig.InterfaceRegistry)
-	ethermint.RegisterInterfaces(encodingConfig.InterfaceRegistry)
+	for _, c := range extraCodecs {
+		if c == "ethermint" {
```
Comment (agouin) [inline client/encoding.go:34]: this should be a switch statement

**Review by agouin — APPROVED** [review]
One nit, otherwise LGTM!

File: client/encoding.go (hunk 1 above)
Comment (joeabbey) [reply client/encoding.go:34]: do we want any protection here to panic if both ethermint AND injective are attempting to be used 🤔

Comment (joeabbey) [reply client/encoding.go:34]: er.. strike that ... the nice thing here is that these are encapsulated in the chains themselves... so unless someone attempts:

```
extra-codecs:
 - ethermint
 - injective
 ```
 
 There should not be an issue.

**Comment by joeabbey** [comment]
fixed #226

**Pull request merged**